presto-main/src/main/java/com/facebook/presto/OutputBuffers.java

```diff
@@ -75,6 +106,13 @@ public boolean isNoMoreBufferIds()
     public void checkValidTransition(OutputBuffers newOutputBuffers)
     {
         requireNonNull(newOutputBuffers, "newOutputBuffers is null");
+        checkArgument(type == newOutputBuffers.getType(), "newOutputBuffers has a different type");
+
+        if (noMoreBufferIds) {
+            checkArgument(this.equals(newOutputBuffers), "Expected buffer to not change after no more buffers is set");
```

Comment:
I just noticed that all callers check the version between calling `checkValidTransition`. This function would just be ignored if the new version is no newer than the original one. Making the following version related check meaningless.